lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -174,6 +186,11 @@ function handleLogs() {
     const region = this.options.region;
     const logGroupName = `/aws/api-gateway/${service}-${stage}`;
 
+    let logFormat = defaultApiGatewayLogFormat;
+    if (typeof logs === 'object' && logs.format) {
```

Comment:
No need for `typeof logs === object`(logically it's dead code).

For any no `null` and no `undefined` value property check is safe (and here we deal with a 100% truthy value, so obviously it's neither a `null` nor `undefined`)

Reply:
But surely `if (logs.format)` may feel as a hacky shortcut (as what is the point to check for `.format` on `true`).

Hence I approved :)

Reply:
Good point! Let's fix it while we're here. Another commit incoming ...